packages/block-directory/src/components/downloadable-blocks-panel/index.js

```diff
@@ -75,19 +75,25 @@ function DownloadableBlocksPanel( {
 
 export default compose( [
 	withSpokenMessages,
-	withSelect( ( select, { filterValue } ) => {
+	withSelect( ( select, { filterValue, rootClientId = null } ) => {
 		const {
 			getDownloadableBlocks,
 			isRequestingDownloadableBlocks,
 		} = select( 'core/block-directory' );
+		const { canInsertBlockType } = select( 'core/block-editor' );
 
 		const hasPermission = select( 'core' ).canUser(
 			'read',
 			'block-directory/search'
 		);
-		const downloadableItems = hasPermission
-			? getDownloadableBlocks( filterValue )
-			: [];
+
+		const searchResultBlocks = getDownloadableBlocks( filterValue ).filter(
```

Comment:
You should transform this into a function and call it only if needed, that means if `hasPermission`.